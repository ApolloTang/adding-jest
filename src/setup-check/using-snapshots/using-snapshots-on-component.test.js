import {getMyData} from './using-snapshots';

import 'react-testing-library/cleanup-after-each';
import {render} from 'react-testing-library';
import React from 'react';
import MyComponent from './using-snapshots-on-component';

test('Using Snapshots',  ()=>{
  const myData = getMyData();
  expect(myData).toMatchSnapshot();
});

test('Using Snapshots on component',  ()=>{
  const {container} = render(<MyComponent />);
  expect(container.innerHTML).toMatchSnapshot();
});

