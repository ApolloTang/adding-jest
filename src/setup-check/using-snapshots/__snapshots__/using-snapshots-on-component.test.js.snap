// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using Snapshots 1`] = `
Array [
  Object {
    "a": "a",
  },
  Object {
    "b": "b",
  },
  Object {
    "c": "c",
  },
]
`;

exports[`Using Snapshots on component.firstChild 1`] = `
<div>
  <div
    class="myStyle"
  >
    <span>
      This is a component is use to check snapshots if work
    </span>
  </div>
</div>
`;

exports[`Using Snapshots on component.innerHTML 1`] = `"<div><div class=\\"myStyle\\"><span>This is a component is use to check snapshots if work</span></div></div>"`;
